### ISSUE TITLE
After nc upgrade to 23.0.4.1, mail demands an update but fails -> disables whole nc

### ISSUE THREAD

**Issue by TobiVanHelsinki** [nextcloud/mail#6383]
### Steps to reproduce

1. be at nc 23.0.3 and click upgrade to 23.0.4
2. run throgh the web based upgrade process
3. get notified, that mail has an important update, that you can't skip (so it seems)
4. click on update, it fails with no reason
5. click on update again, it fails again.
6. be stuck with an update screen. 

### Expected behavior

it is okay if an update fails, but in this case it blocks the complete access to nextcloud

and from my point of view: with no reason. could you overthink this kind of update block?

### Actual behavior

not to be stuck with an update screen that simply says "an error occured"
(or not to be stuck at all)

### Mail app version

1.12

### Operating system

ubuntu 20

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database

MySQL

### Additional info

in cmd there is more information, but is wasn't useful for me. hopefully it will be for you:
```
myncplace$ sudo -u www-data php occ upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Setting log level to debug
Turned on maintenance mode
An unhandled exception has been thrown:
TypeError: Argument 1 passed to Symfony\Component\EventDispatcher\GenericEvent::getArgument() must be of the type string, int given, called in /myncplace/lib/private/EventDispatcher/GenericEventWrapper.php on line 83 and defined in /myncplace/apps/diary/vendor/symfony/event-dispatcher/GenericEvent.php:60
Stack trace:
#0 /myncplace/lib/private/EventDispatcher/GenericEventWrapper.php(83): Symfony\Component\EventDispatcher\GenericEvent->getArgument()
#1 /myncplace/lib/private/Updater.php(499): OC\EventDispatcher\GenericEventWrapper->getArgument()
#2 /myncplace/3rdparty/symfony/event-dispatcher/EventDispatcher.php(251): OC\Updater->OC\{closure}()
#3 /myncplace/3rdparty/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners()
#4 /myncplace/lib/private/EventDispatcher/SymfonyAdapter.php(123): Symfony\Component\EventDispatcher\EventDispatcher->dispatch()
#5 /myncplace/lib/private/Repair.php(260): OC\EventDispatcher\SymfonyAdapter->dispatch()#6 /myncplace/lib/private/Repair.php(123): OC\Repair->emit()
#7 /myncplace/lib/private/Updater.php(252): OC\Repair->run()
#8 /myncplace/lib/private/Updater.php(133): OC\Updater->doUpgrade()
#9 /myncplace/core/Command/Upgrade.php(235): OC\Updater->upgrade()
#10 /myncplace/apps/diary/vendor/symfony/console/Command/Command.php(298): OC\Core\Command\Upgrade->execute()
#11 /myncplace/apps/diary/vendor/symfony/console/Application.php(1015): Symfony\Component\Console\Command\Command->run()
#12 /myncplace/apps/diary/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#13 /myncplace/apps/diary/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#14 /myncplace/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run()
#15 /myncplace/console.php(99): OC\Console\Application->run()
#16 /myncplace/occ(11): require_once('/var/www/html/f...')
```

I ended up with 
`sudo -u www-data php occ app:disable mail`
and now i can access my nc again

**Comment by sargreal** [comment]
Similar problem here, however we are still at Nextcloud 22.x

After Upgrading Mail to the newest available version (v1.12.0) the nextcloud is just "stuck" in the mail update.
The database has high usage, so I suspect there is something going on, but the upgrade command does not complete

**Comment by ChristophWurst** [comment]
> /myncplace/apps/diary/vendor/symfony/console/Command/Command.php

That app should not ship symfony components: https://docs.nextcloud.com/server/latest/developer_manual/app_development/dependency_management.html#dependency-hell
